system-state-dump.sh

```diff
@@ -19,7 +19,10 @@ fi
 echo
 echo
 echo "Applications:"
-for app in /usr/share/applications/*.desktop; do app="${app##/*/}"; echo "${app::-8}"; done
+APPS=$(for app in /usr/share/applications/*.desktop; do app="${app##/*/}"; echo "${app::-8}"; done)
```

Comment:
All `.desktop` entries should have an `Exec=` field that points to en executable available to bash/zsh through `PATH`. 
Pretty sure that `APPS` is always a subset of `IN_PATH` in this case.

Reply:
Counter-example: 
```
cat /usr/share/applications/PyCharm.desktop
[Desktop Entry]
Type=Application
TryExec=/home/bendik/programs/pycharm-2020.2.4/bin/pycharm.sh
Exec=/home/bendik/programs/pycharm-2020.2.4/bin/pycharm.sh
Icon=PyCharm
Categories=TextEditor;IDE;

Name=PyCharm
GenericName=IDE
Comment=IDE for Python development
X-Desktop-File-Install-Version=0.24
```
I don't have `/home/bendik/programs/pycharm-2020.2.4/bin` in my path.

Reply:
I.e. the exec path can be absolute, so doesn't need to be in PATH